### PR TITLE
Button mdx source

### LINKED ISSUE #185: Ensure that the code samples are correct
A lot of stories feature repetitive code. To avoid duplication we generally use functions like
```js
export const heading = (level) => `<h${level}>Heading ${level}</h${level}>`
```
and then we write stories like 
```html
<Preview>
  <Story>{heading(1)}</Story>
</Preview>
```

This eliminates the code repetition but leads to incorrect code previews.
Screen Shot 2020-03-18 at 11 45 35

The correct code can be inserted into the preview using the `mdxSource` prop on the `Preview` component. So a better story would look like
```html
<Preview mdxSource={heading(1)}>
  <Story>{heading(1)}</Story>
</Preview>
```
An example can be found in PR #176.

The goal is to have this done across all stories in the storybook where such function-based code is used.

### PR DESCRIPTION
**Pull request by jahnvigupta**
## Fixes

Related to #185 by @dhruvkb 

## Description

Reduces code and adds mdx source to Button.stories.mdx.
## Checklist
 
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the `master` branch of the repository.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: src/stories/buttons.stories.mdx [hunk 1]
```diff
@@ -5,100 +5,105 @@ import { figmaConfig } from './helpers'
 
 # Buttons
 
+export const buttonClass = (buttonClass) => {
+ return `<button class="${buttonClass}" height="100px">Button</button>`
```
Comment (dhruvkb) [inline src/stories/buttons.stories.mdx]: Your function doesn't produce examples as they were before, go through the old HTML and new HTML carefully to see the difference.

File: src/stories/buttons.stories.mdx [hunk 2]
```diff
@@ -5,100 +5,105 @@ import { figmaConfig } from './helpers'
...
 Buttons are created by adding the `.button` class to a `<button>`.
 
-<Preview>
-  <Story name="MediumButton" height="100px" parameters={{
+<Preview mdxSource={buttonClass('button')}>
+  <Story name="MediumButton" parameters={{
     design: figmaConfig('1403%3A61')
   }}>{`
-    <button class="button">Button</button>
+      ${buttonClass('button')}
```
Comment (dhruvkb) [inline src/stories/buttons.stories.mdx]: Fix indentation.

**Review by dhruvkb — CHANGES_REQUESTED** [review]

**Review by dhruvkb — CHANGES_REQUESTED** [review]
@jahnvigupta have you even seen what you've written here? As far as I can tell, this file contains the same thing twice.

**Comment by jahnvigupta** [comment]
@dhruvkb I am sorry, maybe I just pasted the code without removing the earlier code. Sorry for the inconvenience.

**Review by dhruvkb — APPROVED** [review]

**Pull request merged**